llvm/lib/Target/NVPTX/NVPTXLowerArgs.cpp

```diff
@@ -269,24 +293,44 @@ static void convertToParamAS(Use *OldUse, Value *Param, bool GridConstant) {
                              OriginalUser->getIterator());
         return CvtToGenCall;
       };
-
-      if (auto *CI = dyn_cast<CallInst>(I.OldInstruction)) {
-        I.OldUse->set(GetParamAddrCastToGeneric(I.NewParam, CI));
-        return CI;
+      auto *ParamInGenericAS =
+          GetParamAddrCastToGeneric(I.NewParam, I.OldInstruction);
+
+      // phi/select could use generic arg pointers w/o __grid_constant__
```

Comment:
Ideally, we can also avoid `cvta.param` at Phi/Select uses if Phi/Select's all incoming values have the same address space. However, this pass seems to run before the address space assignment. It could be a further improvement opportunity.

Reply:
Although we always prefer non-generic loads/store, selecting different addr space pointers from phi/select are rare cases. The current implementation should be okay. Just pointed out a potential issue that you may see in the future.